### PR TITLE
Fix structure processing in Connection class

### PR DESCRIPTION
**Pull request by o-rumiantsev**

### CODE REVIEW

File: lib/connection.js [hunk 1]
```diff
@@ -89,13 +100,15 @@ Connection.prototype.sendHandshake = function(handshake) {
 };
 
 Connection.prototype.sendParcel = function(parcel, callback) {
-  const buffer = parser.parcel(parcel, this.encoding);
+  Object.assign(parcel, { encoding: this.encoding });
+  const buffer = parser.parcel(parcel);
```
[the remark this answers is not in the document]
Comment (o-rumiantsev) [reply lib/connection.js]: I think `this.send(parser.parcel(parcel), callback)` is less readable.

**Comment by o-rumiantsev** [comment]
>  What about tests?

@SemenchenkoVitaliy complex tests cant't be implemented while #24 isn't landed.

**Comment by SemenchenkoVitaliy** [comment]
@o-rumiantsev I meant unit tests.

File: lib/connection.js [hunk 2]
```diff
@@ -14,6 +14,15 @@ const ENCODING = {
   v8: 4
 };
 
+const PARCEL_TYPES = [
+  'ping',
```
Comment (bugagashenkj) [inline lib/connection.js]: https://github.com/metarhia/metacom/pull/24#discussion_r326075878

**Comment by tshemsedinov** [comment]
Closed in favour of new version 0.1.x implementing new specification

**Pull request closed without merging**